synapse/util/async_helpers.py

```diff
@@ -485,8 +485,8 @@ def time_it_out():
 
         try:
             deferred.cancel()
-        except:  # noqa: E722, if we throw any exception it'll break time outs
-            logger.exception("Canceller failed during timeout")
+        except BaseException as e:  # if we throw any exception it'll break time outs
```

Comment:
Was `BaseException` always intended to be caught, or should it be `Exception`?

Reply:
These can probably all be `Exception`.